x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/key.test.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { EuiButtonIcon } from '@elastic/eui';
+
+import { Key } from './key';
+
+describe('Key', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const props = {
+    copy: jest.fn(),
+    toggleIsHidden: jest.fn(),
+    isHidden: true,
+    text: '*****',
+  };
+
+  it('renders', () => {
+    const wrapper = shallow(<Key {...props} />);
+    expect(wrapper.find(EuiButtonIcon).length).toEqual(2);
+  });
+
+  it('will call copy when the first button is clicked', () => {
+    const wrapper = shallow(<Key {...props} />);
+    (wrapper.find(EuiButtonIcon).at(0).props() as any).onClick();
```

Comment:
We can simulate a click via Enzyme instead of grabbing on the onClick prop:

```suggestion
    wrapper.find(EuiButtonIcon).first().simulate('click');
```

Reply:
28d1db1dee45684692f640dde28ce023dbfdcee9